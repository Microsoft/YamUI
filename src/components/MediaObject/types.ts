/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
export enum MediaObjectSize {
  XLARGE = 'xLarge',
  LARGE = 'large',
  MEDIUM = 'medium',
  SMALL = 'small',
  XSMALL = 'xSmall',
}
