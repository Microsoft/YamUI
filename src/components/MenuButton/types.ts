/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { ContextualMenuItemType } from 'office-ui-fabric-react/lib/ContextualMenu';

export enum MenuItemType {
  Normal = ContextualMenuItemType.Normal,
  Header = ContextualMenuItemType.Header,
  Divider = ContextualMenuItemType.Divider,
}
