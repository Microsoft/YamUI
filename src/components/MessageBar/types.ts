/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
export enum MessageBarType {
  INFO = 'info',
  WARNING = 'warning',
  ERROR = 'error',
}
