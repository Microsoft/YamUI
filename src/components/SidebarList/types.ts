/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { BaseIllustration } from '../Illustration';

export type SidebarListSize = 'small' | 'medium' | 'large';

export interface BaseSidebarListItem {
  /**
   * A unique key for the item.
   */
  key: string;

  /**
   * The visible title text.
   */
  title: string;

  /**
   * Metadata text displayed under the title, not displayed at small size.
   */
  description?: string;
}
export interface SidebarListIllustrationItem extends BaseSidebarListItem {
  /**
   * A YamUI Illustration component class. You are responsible for importing and providing the correct size.
   */
  illustration: typeof BaseIllustration;
}
export interface SidebarListImageItem extends BaseSidebarListItem {
  /**
   * Custom image JSX. It should fit the list item's pixel width at the given size, or fill 100% of its container width.
   */
  imageContent: React.ReactNode;
}
export type SidebarListItemWithGraphic = SidebarListIllustrationItem | SidebarListImageItem;

export type SidebarListClickableItem = SidebarListItemWithGraphic & {
  /**
   * Callback to be triggered when the item is clicked/invoked.
   */
  onClick(key: BaseSidebarListItem['key']): void;
};
export type SidebarListLinkItem = SidebarListItemWithGraphic & {
  /**
   * The URL this item links to.
   */
  href: string;

  /**
   * If true, opens the link securely in a new window.
   */
  newWindow?: boolean;
};
export type SidebarListItem = SidebarListClickableItem | SidebarListLinkItem;
