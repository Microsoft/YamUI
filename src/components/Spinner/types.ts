/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
export enum SpinnerColor {
  /**
   * Light theme.
   */
  LIGHT = 'light',
  /**
   * Light theme spinner color with gray text.
   */
  METADATA = 'metadata',
  /**
   * Dark theme.
   */
  DARK = 'dark',
}
export enum SpinnerSize {
  LARGE = 'large',
  MEDIUM = 'medium',
  SMALL = 'small',
  XSMALL = 'xSmall',
}
