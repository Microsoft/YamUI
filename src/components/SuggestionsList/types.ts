/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */

import { AvatarProps } from '../Avatar';

export enum SuggestionsListItemTemplate {
  TEXT = 'text',
  AVATAR_ROUND = 'mediaRound',
  MEDIA_SOFT = 'mediaSoft',
  MEDIA_SQUARE = 'mediaSquare',
}

export type CustomizableAvatarProps = 'badgeContent' | 'badgeDescription';

export interface BaseSuggestionItem {
  id: string | number;
  name: string;
}
export interface TextSuggestionItem extends BaseSuggestionItem {
  template: SuggestionsListItemTemplate.TEXT;
}
export interface MediaSuggestionItem extends BaseSuggestionItem {
  template: SuggestionsListItemTemplate.MEDIA_SOFT | SuggestionsListItemTemplate.MEDIA_SQUARE;
  imageUrl?: string;
  description?: string;
}
export interface AvatarSuggestionItem extends BaseSuggestionItem {
  template: SuggestionsListItemTemplate.AVATAR_ROUND;
  imageUrl?: string;
  avatarProps?: Pick<AvatarProps, CustomizableAvatarProps>;
  description?: string;
}
export type SuggestionItem = TextSuggestionItem | MediaSuggestionItem | AvatarSuggestionItem;
