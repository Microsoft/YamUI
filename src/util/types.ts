/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */

export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
