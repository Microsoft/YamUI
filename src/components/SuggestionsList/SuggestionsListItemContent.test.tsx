/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import SuggestionsListItemContent, { SuggestionsListItemContentProps } from './SuggestionsListItemContent';
import { SuggestionItem, SuggestionsListItemTemplate } from './types';

describe('<SuggestionsListItemContentContent />', () => {
  let component: ShallowWrapper<SuggestionsListItemContentProps>;

  const textItem: SuggestionItem = {
    template: SuggestionsListItemTemplate.TEXT,
    id: '1',
    name: 'John Smith',
  };
  const avatarItem: SuggestionItem = {
    template: SuggestionsListItemTemplate.AVATAR_ROUND,
    id: '1',
    name: 'John Smith',
    description: 'description',
    imageUrl: 'user.png',
    avatarProps: { badgeContent: 'badge' },
  };
  const mediaSoftItem: SuggestionItem = {
    template: SuggestionsListItemTemplate.MEDIA_SOFT,
    id: '1',
    name: 'John Smith',
    description: 'description',
    imageUrl: 'user.png',
  };
  const mediaSquareItem: SuggestionItem = {
    template: SuggestionsListItemTemplate.MEDIA_SQUARE,
    id: '1',
    name: 'John Smith',
    description: 'description',
    imageUrl: 'user.png',
  };
  const mediaSquareItemWithoutImage: SuggestionItem = {
    template: SuggestionsListItemTemplate.MEDIA_SQUARE,
    id: '1',
    name: 'John Smith',
    description: 'description',
  };

  describe('with text item', () => {
    beforeEach(() => {
      component = shallow(<SuggestionsListItemContent item={textItem} />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with avatar item', () => {
    beforeEach(() => {
      component = shallow(<SuggestionsListItemContent item={avatarItem} />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with media soft item', () => {
    beforeEach(() => {
      component = shallow(<SuggestionsListItemContent item={mediaSoftItem} />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with media square item', () => {
    beforeEach(() => {
      component = shallow(<SuggestionsListItemContent item={mediaSquareItem} />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with media item without image', () => {
    beforeEach(() => {
      component = shallow(<SuggestionsListItemContent item={mediaSquareItemWithoutImage} />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with search text', () => {
    beforeEach(() => {
      component = shallow(<SuggestionsListItemContent item={avatarItem} searchText="Joh" />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
