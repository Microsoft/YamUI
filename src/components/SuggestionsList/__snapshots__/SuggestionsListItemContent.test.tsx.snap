// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuggestionsListItemContentContent /> with avatar item matches its snapshot 1`] = `
<div
  className=
      y-suggestionsListItem
      {
        padding-bottom: 8px;
        padding-left: 12px;
        padding-right: 12px;
        padding-top: 4px;
        text-align: left;
      }
>
  <MediaObject
    imageContent={
      <CustomizedAvatar
        badgeContent="badge"
        imageUrl="user.png"
        name="John Smith"
        size="small"
      />
    }
    metadataContent="description"
    size="small"
    titleContent="John Smith"
  />
</div>
`;

exports[`<SuggestionsListItemContentContent /> with media item without image matches its snapshot 1`] = `
<div
  className=
      y-suggestionsListItem
      {
        padding-bottom: 8px;
        padding-left: 12px;
        padding-right: 12px;
        padding-top: 4px;
        text-align: left;
      }
>
  <MediaObject
    imageContent={null}
    metadataContent="description"
    size="small"
    titleContent="John Smith"
  />
</div>
`;

exports[`<SuggestionsListItemContentContent /> with media soft item matches its snapshot 1`] = `
<div
  className=
      y-suggestionsListItem
      {
        padding-bottom: 8px;
        padding-left: 12px;
        padding-right: 12px;
        padding-top: 4px;
        text-align: left;
      }
>
  <MediaObject
    imageContent={
      <span
        aria-hidden={true}
      >
        <Image
          block={true}
          borderType="soft"
          description=""
          fullWidth={false}
          height={32}
          imageFit={2}
          shouldFadeIn={false}
          source="user.png"
          width={32}
        />
      </span>
    }
    metadataContent="description"
    size="small"
    titleContent="John Smith"
  />
</div>
`;

exports[`<SuggestionsListItemContentContent /> with media square item matches its snapshot 1`] = `
<div
  className=
      y-suggestionsListItem
      {
        padding-bottom: 8px;
        padding-left: 12px;
        padding-right: 12px;
        padding-top: 4px;
        text-align: left;
      }
>
  <MediaObject
    imageContent={
      <span
        aria-hidden={true}
      >
        <Image
          block={true}
          description=""
          fullWidth={false}
          height={32}
          imageFit={2}
          shouldFadeIn={false}
          source="user.png"
          width={32}
        />
      </span>
    }
    metadataContent="description"
    size="small"
    titleContent="John Smith"
  />
</div>
`;

exports[`<SuggestionsListItemContentContent /> with search text matches its snapshot 1`] = `
<div
  className=
      y-suggestionsListItem
      {
        padding-bottom: 8px;
        padding-left: 12px;
        padding-right: 12px;
        padding-top: 4px;
        text-align: left;
      }
>
  <MediaObject
    imageContent={
      <CustomizedAvatar
        badgeContent="badge"
        imageUrl="user.png"
        name="John Smith"
        size="small"
      />
    }
    metadataContent="description"
    size="small"
    titleContent={
      Array [
        <span
          className=

              {
                font-weight: 600;
              }
        >
          Joh
        </span>,
        <span>
          n Smith
        </span>,
      ]
    }
  />
</div>
`;

exports[`<SuggestionsListItemContentContent /> with text item matches its snapshot 1`] = `
<div
  className=
      y-suggestionsListItem
      {
        padding-bottom: 4px;
        padding-left: 12px;
        padding-right: 12px;
        padding-top: 4px;
        text-align: left;
      }
>
  <CustomizedBlock
    textColor="primary"
    textSize="mediumSub"
  >
    John Smith
  </CustomizedBlock>
</div>
`;
